### PR TITLE
Make fab auto-create tasks for each env

### PR DESCRIPTION
**Pull request by dannyroberts**
Move any differences between envs into `app-settings.yml` so that creating the `@task` for an env is rote; then just auto-generate those tasks. Removes a lot of boilerplate code and puts these differences together into a config file in the environment.

### CODE REVIEW

File: fab/fab/fabfile.py [hunk 1]
```diff
@@ -167,107 +160,59 @@ def get_env_dict(path):
...
+    if window:
+        d = datetime.datetime.now(pytz.timezone(window['timezone']))
+        if window['hour_start'] <= d.hour < window['hour_end']:
+            return
+    else:
         return
 
     message = (
-        "Woah there bud! You're deploying '%s' during the day. "
+        "Whoa there bud! You're deploying '%s' during the day. "
```
Comment (czue) [inline fab/fab/fabfile.py]: [the more you know](https://www.grammarly.com/blog/whoa-woah/)

Comment (dannyroberts) [reply fab/fab/fabfile.py]: It apparently even has a [known etymology](https://www.etymonline.com/word/whoa). You learn interesting things as a programmer.

Comment (emord) [reply fab/fab/fabfile.py]: whoa

**Pull request merged**